mathics/builtin/logic.py

```diff
@@ -3,7 +3,70 @@
 from mathics.version import __version__  # noqa used in loading to check consistency.
 from mathics.builtin.base import BinaryOperator, Predefined, PrefixOperator, Builtin
 from mathics.builtin.lists import InvalidLevelspecError, python_levelspec, walk_levels
-from mathics.core.expression import Expression, Symbol
+from mathics.core.expression import (
+    Expression,
+    Symbol,
+    SymbolTrue,
+    SymbolFalse,
+    SymbolUndefined,
+    SymbolList,
+)
+
+from mathics.core.rules import Rule
+
+from typing import Optional, Union, Any
+
+
+def evaluate_predicate(pred, evaluation):
+    assumptions_list = get_assumptions_list(evaluation)
+    assumption_rules = []
+    for pat in assumptions_list:
+        if pat.head == Symbol("Not"):
```

Comment:
is_form again here?

Reply:
the same: we do not know a priori if pat is a Symbol or an Expression. This is the safer and simpler way to do the comparison.